### ISSUE TITLE
[BUG] - Using BoxAnnotation with Jupyter output does not work properly

### ISSUE THREAD

**Issue by LegrandNico** [bokeh/bokeh#11448]
Adding BoxAnnotation create a bug when using jupyter output (but not HTML).

I am using the following software:

Python 3.7
jupyterlab 3.0.3
bokeh 2.3.3
Firefox last version
Windows 10

The following code should reproduce the issue. 

```python
import numpy as np
import pandas as pd
from bokeh.models import Circle, Line, BoxAnnotation
from bokeh.models.tools import HoverTool
from bokeh.plotting import ColumnDataSource, figure
from bokeh.plotting.figure import Figure
from bokeh.plotting import show
from bokeh.io import output_notebook
output_notebook()


p1 = figure(
    title="Test",
    sizing_mode="stretch_width",
    plot_height=400,
    tools="pan",
)

# Data
hr, time = np.random.normal(0, 1, 1000), np.arange(0, 1000)
time = pd.to_datetime(time, unit="s", origin="unix")
line_source = ColumnDataSource(data=dict(time=time, hr=hr, bpm=60000 / hr))

linePlot = Line(
    x="time",
    y="hr",
    line_color="#4c72b0",
)
p1.add_glyph(
    line_source,
    linePlot,
)

# Adding box annotation will reproduce the error
high, low = (3000, 200)
upper_bound = BoxAnnotation(bottom=1, fill_alpha=0.1, fill_color='red')
p1.add_layout(upper_bound)

show(p1)

```
The unexpected behaviour only appears when you start to use the pan and move the plot (or use another kind of hover or interactive tool in general). Then it looks like many plots are stacked on the top of each other (which might be the case as the plot itself get laggy and slow). 

<img width="1151" alt="bokeh_boxannotation_error" src="https://user-images.githubusercontent.com/27711796/127157621-5b6f9e88-d4b4-4613-9653-dd5512f1b6df.png">

**Comment by mattpap** [comment]
The image says that you are actually using bokeh 2.3.0 and hitting [this](https://github.com/bokeh/bokeh/issues/11054) bug (or its variant), instead of the reported 2.3.3. This was fixed in bokeh 2.3.1. Upgrade to the most recent version and the problem should go away.

**Comment by LegrandNico** [comment]
Yes indeed, I had to restart Jupyter-lab to load the last Bokeh version -_- 

This is working now, sorry for the inconvenience.
